dev/bazel/deps/mpi.bzl

```diff
@@ -0,0 +1,104 @@
+#===============================================================================
+# Copyright 2020-2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+load("@onedal//dev/bazel:repos.bzl", "repos")
+
+mpi_repo = repos.prebuilt_libs_repo_rule(
+    includes = [
+        "include",
+    ],
+    libs = [
+        "lib/release/libmpi.so",
+        "lib/release/libmpi.so.12",
+        "lib/release/libmpi.so.12.0",
+        "lib/release/libmpi.so.12.0.0",
+        "libfabric/lib/libfabric.so",
+        "libfabric/lib/libfabric.so.1",
+        "libfabric/lib/prov/libefa-fi.so",
+        "libfabric/lib/prov/libmlx-fi.so",
+        "libfabric/lib/prov/libpsmx2-fi.so",
+        "libfabric/lib/prov/librxm-fi.so",
+        "libfabric/lib/prov/libshm-fi.so",
+        "libfabric/lib/prov/libsockets-fi.so",
+        "libfabric/lib/prov/libtcp-fi.so",
+        "libfabric/lib/prov/libverbs-fi.so",
+    ],
+    build_template = "@onedal//dev/bazel/deps:mpi.tpl.BUILD",
+    download_mapping = {
+        # Required directory layout and layout in the downloaded
+        # archives may be different. Mapping helps to setup relations
+        # between the required layout (LHS) and downloaded (RHS).
+        #          REQUIRED                       DOWNLOADED
+        "libfabric/lib/libfabric.so":    "lib/libfabric/libfabric.so",
+        "libfabric/lib/libfabric.so.1":  "lib/libfabric/libfabric.so.1",
+        "lib/release/libmpi.so":         "lib/libmpi.so",
+        "lib/release/libmpi.so.12":      "lib/libmpi.so.12",
+        "lib/release/libmpi.so.12.0":    "lib/libmpi.so.12.0",
+        "lib/release/libmpi.so.12.0.0":  "lib/libmpi.so.12.0.0",
+        "libfabric/lib/prov":            "lib/libfabric/prov",
+    },
+)
+
+def _get_fi_providers_dir(fi_files):
+    if len(fi_files) == 0:
+        fail("No fabrin interface files provided for MPI")
+    fi_dir = fi_files[0].dirname
+    for fi in fi_files:
+        if fi.dirname != fi_dir:
+            fail("All fabric interface files must reside in the same directory")
+    return fi_dir
+
+def _generate_exec_wrapper(ctx, actions, executable, fi_dir):
+    exec_wrapper = actions.declare_file(executable.basename + ".sh")
+    content = (
+        "#!/bin/bash\n" +
```

Comment:
Can we use smth like Jinja (or any other templating engine)?

Reply:
No, Jinja is Python, there is no Python deps here. It looks like a python, but not Python